### ISSUE TITLE
URL to check token badly constructed

### ISSUE THREAD

**Issue by tjamme** [certbot/certbot#5238]
## My operating system is (include version):
(Ubuntu 4.8.4-2ubuntu1~14.04.3

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
Downloaded from site

## I ran this command and it produced this output:
./certbot-auto --manual -d footprint.wwf.org.uk -d www.footprint.wwf.org.uk certonly

Problem is: MISSING FORWARD SLASH after .org.uk
http://footprint.wwf.org.uk.well-known/acme-challenge/OOi-aLYqH3eh7VUo-JDGhKOKAL_4hZfXwR7AKJuGArI:http://footprint.wwf.org.uk.well-known/acme-challenge/OOi-aLYqH3eh7VUo-JDGhKOKAL_4hZfXwR7AKJuGArI:
Error getting validation data <===

## Certbot's behavior differed from what I expected because:
It gave me an unjustified error: if it out the URL together wrong, it wasn't going to find the token...

**Comment by stAn47** [comment]
i get the same problem with .software domain 

Domain: onepagecheckout.software
Type:   connection
Detail: Fetching
https://onepagecheckout.software.well-known/acme-challenge/6PE4N_6Q6v_RbjqokspnY39FWy2y5MI-TMwRcKQaAfY:
Error getting validation data

**Comment by SwartzCr** [comment]
@stAn47 Can you share the command you ran? Also what OS are you on?
@tjamme would you be able to share the whole log for the run of Certbot?

**Comment by joohoi** [comment]
@stAn47 looks like your site has some kind of redirection, either in the web application or httpd configuration:
```
▶ curl -v http://onepagecheckout.software/.well-known/acme-challenge/whatever
*   Trying 92.240.237.203...
* TCP_NODELAY set
* Connected to onepagecheckout.software (92.240.237.203) port 80 (#0)
> GET /.well-known/acme-challenge/whatever HTTP/1.1
> Host: onepagecheckout.software
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 302 Found
< Date: Tue, 14 Nov 2017 23:05:03 GMT
< Server: Apache/2.4.29
< Location: https://onepagecheckout.software.well-known/acme-challenge/whatever
< Content-Length: 251
< Connection: close
< Content-Type: text/html; charset=iso-8859-1
< 
```

I couldn't find similar issue with domains reported by @tjamme but Let's Encrypt seems to report back whatever URL was the last one it tried in a chain of HTTP redirects. Could you verify that this is not the case? Better yet, could you leave the validation token on your host for debugging?

**Comment by tjamme** [comment]
Riiight.... We were missing a trailing slash in the redirection.
So the concatenation is actually done by the server.
I was going to moan that letsencrypt SHOULD check it has a / before its .wellknown folder, but it is conceivable that some redirections may redirect that folder to something unpredictable.

Given that when I got my search terms right I found thousands of people on Google with the same problem, it would help the world at large to make the issue prominent in the doc or to have it mentioned when a redirection is detected and the host could not be found. Just saying.
T.
